### DISCUSSION TITLE
Support Nitro Enclaves

### DISCUSSION THREAD

**Discussion by bwagner5** [aws/karpenter-provider-aws#8962] [Feature Requests]
### Tell us about your request

Support AWS Nitro Enclaves - https://aws.amazon.com/ec2/nitro/nitro-enclaves/

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

Use Nitro Enclaves w/ Karpenter

### Are you currently working around this issue?

No

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by sftim** [comment]
I think this use case would be effectively served through a combination of:
- [Dynamic resource allocation](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
- Karpenter node management

I can see how its important; however, getting the design right (or even good enough to not have frustrations) will take some work.

**Comment by mlschindler** [comment]
My org is very interested in seeing this come to fruition. One of the issues we run into with Nitro Enclaves is random page size allocations; we are supposed to be able to guarantee any arbitrary Huge Page size, but AWS doesn't respect it properly on node provision. We have multiple open tickets with them and I can link findings back here when it becomes appropriate.

**Comment by james-callahan** [comment]
For what it's worth, for our nitro enclave usage the main thing missing is the custom resource support, i.e. https://github.com/aws/karpenter-core/issues/751

**Comment by sergiy-slobodyan** [comment]
Converted to a discussion for now. Please see the reasoning in this reply

https://github.com/aws/karpenter-provider-aws/pull/8165#issuecomment-3746609646
